### PR TITLE
Decision Layer unit tests + CI

### PR DESCRIPTION
**Pull request by bruvduroiu**
Adds Unit Tests for the `DecisionLayer` class and Encoders.

Sets up Github Workflow pipeline for `pytest` unit tests + code coverage report

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -14,12 +15,17 @@ python = "^3.10"
 pydantic = "^1.8.2"
 openai = "^0.28.1"
 cohere = "^4.32"
+numpy = "^1.26.2"
+scipy = "^1.11.4"
```
Comment (bruvduroiu) [inline pyproject.toml:19]: Seems like these libraries were missing, though being used within the project

Comment (jamescalam) [reply pyproject.toml:19]: `numpy` should be included for sure — I don't know where `scipy` is being used though?

**Review by jamescalam — CHANGES_REQUESTED** [review]
looks good to me, just need to run linting with `poetry run make lint` or `make format`

File: pyproject.toml (hunk 1 above)
Comment (bruvduroiu) [reply pyproject.toml:19]: Ok, nvm I got confused, I see we're actually using `numpy.linalg.norm` not the scipy version

File: .python-version [hunk 2]
```diff
@@ -0,0 +1 @@
+3.11
```
Comment (bruvduroiu) [inline .python-version:1]: Good idea to explicitly define our Python version for contributors that use `pyenv` to manage Python environments.

Comment (jamescalam) [reply .python-version:1]: good idea

**Pull request merged**